### PR TITLE
:bug: Add fix for issue2277

### LINKED ISSUE #2277: BUG Running Scorecard on a private repo always results in a score of 10 for Token-Permissions and Pinned-Dependencies check
**Bug Description**
If `scorecard` is run on a `private-repo`  with `Token-Permissions or  Pinned-Dependencies` checks. The score is not calculated properly. It always shows `max-score` i.e 10. Whereas if you run the same checks on a public repo, the score is calculated correctly.

**Reproduction steps**
> Pre-requisite: A private-repo with a workflow file having no token-permission and no pinned actions

1. Download the latest version of scorecard
2. Run the below command.
```bash
GITHUB_AUTH_TOKEN="<tokenHere>" ./scorecard-linux-amd64 --repo "owner/repo" --checks=Token-Permissions,Pinned-Dependencies --verbosity Debug
```
3. Observe the score.
4. Change the visibility of same repo to `public`.
5. Re-run the above command.
6. Observe the score again.

By performing the above steps; you will observe the difference in score.

**Expected behavior**
For both runs; the score should be same

----

### Current Flow of Scorecards for calculating score of pinning & tokenPermission
1. Fetch the tarball, of the repo
2. Perform the analysis, on `Dockerfile` or `Workflow` files for `pinned-Dependencies or Token-Permission`
3. Generate Score.

### Cause of Bug

Authorization Token is not supplied in `http-call` for fetching `tarball`. The calls fails for `private-repos` and works for `public-repos`, because of `authorizationToken not being supplied`.

### Place of Bug.
The miscalculation of score occurs in `step2 & step3`; if `fetching of tarball fails`. 
In current code for `fetching tarball` [here](https://github.com/ossf/scorecard/blob/10b6052acfb4f0b8136bc4876cb6f5b6f26bfe93/clients/githubrepo/tarball.go#L113), if you see the preparation of request at `line 121`; there is no `Authorization Header` being added due to which the request fails for `private-repos`, which causes miscalculation of scores.

### Fix
In order to fix this bug;
* I tried adding `authorizationHeader` in my [fork](https://github.com/h0x0er/scorecard/blob/main/clients/githubrepo/tarball.go#L125); 
* build the binary in local machine and 
* Ran it.
The outcome was `positive`, the changes worked :)

/cc @varunsh-coder @boahc077

### PR DESCRIPTION
**Pull request by h0x0er**
#### What kind of change does this PR introduce?
This PR adds a `httpClient`  in `tarballHandler` that adds `authorizationHeader` while sending requests; because the current way of fetching tarball fails for `private-repos`  as no `Authorization Header` is being added

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### Which issue(s) this PR fixes
Fixes #2277

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: clients/githubrepo/tarball.go [hunk 1]
```diff
@@ -80,6 +83,7 @@ func (handler *tarballHandler) init(ctx context.Context, repo *github.Repository
 	handler.once = new(sync.Once)
 	handler.ctx = ctx
 	handler.repo = repo
+	handler.httpClient = makeHTTPClient(ctx)
```
Comment (azeemshaikh38) [inline clients/githubrepo/tarball.go:86]: Can simply pass in `httpClient` here - https://github.com/ossf/scorecard/blob/main/clients/githubrepo/client.go#L233: 

```
httpClient := ...
return &Client {
  tarball: tarballHandler{
    httpClient: httpClient,
  },
  ...
}
```

**Review by azeemshaikh38 — APPROVED** [review]
Thanks! Approved with comment.

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by azeemshaikh38** [comment]
@h0x0er please sign your commits, see - https://github.com/ossf/scorecard/pull/2278/checks?check_run_id=8454801894

**Review by laurentsimon — APPROVED** [review]
nice catch, thanks

/cc @azeemshaikh38

**Pull request merged**